src/Entity/EntityDocument.php

```diff
@@ -31,4 +33,15 @@ public function getId();
 	 */
 	public function getType();
 
+	/**
+	 * Sets the id of the entity. If the id is not of the correct type,
+	 * an exception will be thrown. A specific derivative of EntityId is
+	 * always supported.
+	 *
+	 * @param EntityId|mixed $id
```

Comment:
There is no interpretation of what `mixed` means, it means "all types". So whatever you do, no matter if you start with `EntityId` and restrict it to a subtype or if you start with `mixed` and restrict it to `ItemId|int|null`, it will always be a LSP violation. Which makes me wonder. Is there even a "correct" way to solve this?